L65-python-package-name.md

```diff
@@ -0,0 +1,132 @@
+Additional PyPI Packages for gRPC Python
+----
+* Author(s): lidiz
+* Approver: gnossen
+* Status: Draft
+* Implemented in: python
+* Last updated: 05/13/2020
+* Discussion at: **TBD**
+
+## Abstract
+
+gRPC Python is uploaded as `grpcio` on PyPI, but there is another package named
+`grpc`. Hence some users are confused. This document proposes gRPC Python to
+upload additional packages named `grpc` and `grpc-*` as proxies to our official
+packages.
+
+## Background
+
+### Package `grpc`
+
+Package `grpc` is an alternative version of gRPC Python only supports Python 2,
+and the first version is uploaded in 2013. We have seen complaints about the
+installation failure or unable to execute our example code. The tricky package
+name is a usability issue for years.
+
+### Package `grpcio`
+
+Package `grpcio` is first uploaded in 2015, given the fact that the name `grpc`
+already being taken, this name is likely a workaround.
+
+## Proposal
+
+### List of Package Name
+
+Here is a list of new proxy packages:
+* `grpc`
+* `grpc-status`
+* `grpc-channelz`
+* `grpc-tools`
+* `grpc-reflection`
+* `grpc-testing`
+* `grpc-health-checking`
+
+### Proxy Mechanism
```

Comment:
"Proxy" isn't the right name for what we're doing here. What we're really doing is nerfing the `grpc` package with a message suggesting that they install `grpcio` instead. This mechanism is the sign in this picture: ![](https://i.imgur.com/mSHi8.jpg)

Whereas a proxy would actually be a cop.

Reply:
How should we call it? "Cop packages" / "Guard packages" / "Watchdog packages" / "Toxic packages"? What do you recommend?

Reply:
No. I didn't mean things quite so literally with the meme. "Archived package" or "retired package" is probably best.

Reply:
It's sad that those new packages never shined, but went straight to retirement. Perhaps that's been the story of life. Could it be better to reference it as just "package", and avoid using the word "proxy"?

Reply:
They did have a chance to shine. As "gevent RPC". *That's* the package that I'm referring to as archived, not a hypothetical "gRPC" `grpc` package.

I'm fine referring to them as just "the `grpc`-prefixed packages" or something like that. So long as it's clear in each instance whether we're referring to `grpcio`, `grpcio-reflection`, etc. *or* `grpc`, `grpc-testing`, etc.